src/health/check/diesel.rs

```diff
@@ -0,0 +1,51 @@
+use ::diesel::Connection;
+
+use crate::health::{self};
+
+use super::Diesel;
+
+impl<C: Connection, T: (Fn() -> Result<C, String>) + Sync + Send> Diesel<C, T> {
+    /// Creates a new diesel health indicator using "select 1" as query.
+    pub fn new(name: String, connection_provider: T) -> Self {
+        Self {
+            name,
+            query: "select 1".to_string(),
+            connection_provider,
+        }
+    }
+
+    /// Creates a new diesel health indicator with a custom query.
+    pub fn new_with_query(name: String, query: String, connection_provider: T) -> Self {
+        Self {
+            name,
+            query,
+            connection_provider,
+        }
+    }
+}
+
+#[async_trait::async_trait]
+impl<C: Connection, T: (Fn() -> Result<C, String>) + Sync + Send> health::Indicator
+    for Diesel<C, T>
+{
+    fn name(&self) -> &str {
+        &self.name
+    }
+
+    async fn check(&self) -> Result<(), String> {
+        let mut conn = (self.connection_provider)()?;
+
+        conn.batch_execute(&self.query).map_err(|e| e.to_string())?;
```

Comment:
wouldn't executing direct string comments cause security issues 🤔

Reply:
I don't think this should be a security issue. Here, the query is not taken from an external user input and is not dynamically created. We can expect the library user to take any necessary security measures. I find this saying something like "I can execute a string query using Diesel, therefore Diesel is not secure".